### PR TITLE
fix: sealing: Set all path types for Unseal pipeline to sealing storage

### PR DESCRIPTION
**Pull request by shrenujbansal**
## Related Issues

https://github.com/filecoin-project/lotus/issues/9159

## Proposed Changes

Currently when unsealing a sector, its expected that the unsealing worker has access to long term storage and stores the unsealed data there. We need to be able to unseal data and deliver it to the miner without having access to long term storage for retrievals.
Instead of using long term storage, use sealing storage to in the unsealing pipeline

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] New features have usage guidelines and / or documentation updates in
  - [x] [Lotus Documentation](https://lotus.filecoin.io)
  - [x] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

File: itests/sector_unseal_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	require.NoError(t, err)
+
+	sectorRef := storiface.SectorRef{
+		ID:        abi.SectorID{Miner: abi.ActorID(minerId), Number: sector},
+		ProofType: sinfo.SealProof,
+	}
+
+	err = miner.SectorsUnsealPiece(ctx, sectorRef, 0, 0, sinfo.Ticket.Value, sinfo.CommD)
+	require.NoError(t, err)
+}
```
Comment (magik6k) [inline itests/sector_unseal_test.go:90]: Can we add some asserts that:
* The Unsealed file is in long-term storage
* Sealed/Cache still are in long-term storage
* There is nothing left in sealing storage

**Review by magik6k — COMMENTED** [review]
I'd add some asserts to the test to make sure this does what we expect

File: itests/sector_unseal_test.go (hunk 1 above)
Comment (shrenujbansal) [reply itests/sector_unseal_test.go:142]: done

**Comment by hyunmoon** [comment]
I can confirm that this issue is not fixed in v1.19.0+mainnet

**Comment by hyunmoon** [comment]
But I'm glad to know that I can work around the issue by letting the workers have access to the long term storage.

**Pull request merged**